src/utilstrencodings.cpp

```diff
@@ -123,6 +125,75 @@ vector<unsigned char> ParseHex(const string& str)
     return ParseHex(str.c_str());
 }
 
+string EncodeAscii85(const char* istr, size_t len)
+{
+    
+    string retVal; //Default is empty-string
+    if (NULL != istr)
+    {
+        size_t isz = strlen(istr);
+        int32_t olen = ascii85_get_max_encoded_length(isz);
+        
+        if (olen < 0)
+        {
+            return retVal; //Encode size error
+        }
+        else
+        {
+            uint8_t obuf[olen];
+            olen = encode_ascii85((uint8_t *)istr, isz, obuf, olen);
+            if(0 < olen)
+            {
+                std::ostringstream ss;
+                std::copy(obuf, obuf+olen, std::ostream_iterator<uint8_t>(ss, ""));
+                retVal = ss.str();
+            }
+        }
+    }
+    return retVal;
+}
+
+string EncodeAscii85(const string& str)
+{
+    return EncodeAscii85((const char*)str.c_str(), str.size());
+}
+
+vector<unsigned char> DecodeAscii85(const char* ostr, bool* pfInvalid)
+{
+    vector<unsigned char> retVal;
+
+    if (NULL != ostr)
+    {
+        size_t osz = strlen(ostr);
+        int32_t olen = ascii85_get_max_decoded_length(osz);
+        if (olen < 0)
+        {
+            *pfInvalid = true;//Decode size error
+            return retVal;
+        }
+        else
+        {
+            uint8_t dbuf[olen];
+            int32_t ilen = decode_ascii85((uint8_t *)ostr, osz, dbuf, olen);
+
+            if (ilen < 0)
+            {
+                *pfInvalid = true;//Decode error
+                return retVal;
+            }
+            retVal = std::vector<unsigned char>(dbuf, dbuf + ilen);     
```

Comment:
this is again a double copy. Here you creating vector object from dbuf, and then making a copy to retVal. 
just use retVal.assign(dbuf, dbuf + ilen);